src/status_im/chat/views/message/options.cljs

```diff
@@ -0,0 +1,49 @@
+(ns status-im.chat.views.message.options
+  (:require-macros [status-im.utils.views :refer [defview]])
+  (:require [re-frame.core :as re-frame]
+            [reagent.core :as reagent]
+            [clojure.string :as string]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.colors :as colors]
+            [status-im.chat.views.bottom-info :as bottom-info]
+            [status-im.chat.styles.screen :as styles]
+            [status-im.chat.styles.message.options :as options.styles]
+            [status-im.i18n :as i18n]
+            [status-im.ui.components.animation :as anim]
+            [status-im.ui.components.list.views :as list]
+            [status-im.utils.core :as utils]
+            [status-im.utils.identicon :as identicon]
+            [status-im.ui.components.icons.vector-icons :as vector-icons]))
+
+(defn action-item [{:keys [label icon style on-press]}]
+  [react/touchable-highlight {:on-press on-press}
+   [react/view options.styles/row
+    [react/view
+     [vector-icons/icon icon style]]
+    [react/view (merge options.styles/label style)
+     [react/text {:style (merge options.styles/label-text style)} (i18n/label label)]]]])
+
+(defn view []
+  (let [message-options (re-frame/subscribe [:get-current-chat-ui-prop :message-options])
+        close-message-options-fn #(re-frame/dispatch-sync [:set-chat-ui-props {:show-message-options? false}])]
+    (reagent/create-class
```

Comment:
Sorry if this has been discussed before, but when I see `reagent/create-class` I think that we are using some of the lifecycle methods (`component-did-update`,`component-will-unmount`..), here we are only using `display-name`, is that the reason for using it? Should we always use it instead? ( I know `defview` sets it, although it is based on the function name)

Reply:
Right, no need to use `reagent/create-class`, will remove.